File: app/src/main/java/io/github/rebirthlee/spinnermvvm/SimpleItemViewModel.java
package io.github.rebirthlee.spinnermvvm;

public class SimpleItemViewModel {
    private Person model;
    private OnClickListener<Person> listener;

    public SimpleItemViewModel(Person model, OnClickListener<Person> listener) {
        this.model = model;
        this.listener = listener;
    }

    public void onClick() {
        listener.onClick(model);
    }

    public String getUrl() {
        return model.url;
    }

    public void setUrl(String url) {
        model.url = url;
    }

    public String getTitle() {
        return model.name;
    }

    public void setTitle(String title) {
        model.name = title;
    }
}

File: kotlinApp/src/main/java/io/github/rebirthlee/spinnermvvm/kotlin/SimpleItemViewModel.kt
package io.github.rebirthlee.spinnermvvm.kotlin

class SimpleItemViewModel(private val model: Person, private val listener: (model: Person) -> Unit) {

    var url: String
        get() = model.url
        set(url) {
            model.url = url
        }

    var title: String
        get() = model.name
        set(title) {
            model.name = title
        }

    fun onClick() {
        listener(model)
    }
}
